src/lib/mongoBackend/mongoSubCache.cpp

```diff
@@ -251,7 +251,17 @@ int mongoSubCacheItemInsert(const char* tenant, const BSONObj& sub)
     }
 
     std::string id            = getStringFieldF(entity, ENT_ENTITY_ID);
-    std::string isPattern     = entity.hasField(CSUB_ENTITY_ISPATTERN)? getStringFieldF(entity, CSUB_ENTITY_ISPATTERN) : "false";
+    std::string isPattern;
+    if (entity.hasField(CSUB_ENTITY_ISPATTERN))
+    {
+      isPattern = getStringFieldF(entity, CSUB_ENTITY_ISPATTERN);
```

Comment:
Please use the hasField before actually getting the value.
If the field isn't present, the broker may crash ...

Reply:
I appreciate your comment. Based on your comment, I try to fix the code more simply and with less impact on the original code as below.
```
std::string isPattern     = entity.hasField(CSUB_ENTITY_ISPATTERN)? getStringFieldF(entity, CSUB_ENTITY_ISPATTERN) : "false";
if (isPattern.empty() && id.empty()) 
{
  isPattern = "true";
}
```

Reply:
Wait, I was too fast with my comment.
You already check that isPattern is present (on the line above).
The first line in your comment will be perfect:
```
std::string isPattern     = entity.hasField(CSUB_ENTITY_ISPATTERN)? getStringFieldF(entity, CSUB_ENTITY_ISPATTERN) : "false";
```
the following lines we don't need  `if (isPattern.empty() && id.empty()) {  isPattern = "true"; })` - remove those.

It was good already. With the new proposal it's done on 4 lines instead of 1.

Reply:
The line `if (isPattern.empty() && id.empty()) { isPattern = "true"; }` is a key point of my proposed bug fix.
If the subscription does not designate both `id` and `isPattern`, the subscription is stored in Mongo DB as `{ id: "", isPattern: "", ... }`.
In this case, `mongoSubCacheItemInsert` set `id` as empty string and `isPattern` as `false`. Then it does not match any entities in `EntityInfo::match` function of `subCache.cpp`, and the notification does not work. 
Therefore, `mongoSubCacheItemInsert` should set `isPattern` as `true` if both `id` and `isPattern` are empty.

Reply:
Ah ok, I was too fast again - you merged the two modification "into one".
Apologies.
ok - good - do that.

If only entity type is given - isPattern must be true, of course. But, the id should be set to '.*', otherwise I'm not sure it will work

Reply:
OK. Thank you for your suggestion.
I will fix the code as below.
```
if (isPattern.empty() && id.empty()) 
{
  id = ".*";
  isPattern = "true";
}
```

Reply:
Great